components/shared/onboarding/Onboarding.tsx

```diff
@@ -0,0 +1,320 @@
+    handleSubmit,
+    control,
+    register,
+    getValues,
+    formState: { errors },
+  } = useForm<z.infer<typeof CompleteProfileEditSchema>>({
+    resolver: zodResolver(CompleteProfileEditSchema),
+    defaultValues: {
+      learningGoals: user.learningGoals || [],
+      experiences: user.experiences?.map((exp) => ({ name: exp })) || [],
+      availability: {
+        startTime: user?.startTime || new Date(),
+        endTime: user?.endTime || new Date(),
+      },
+      newProjects: user?.newProjects || false,
+      technologies: user?.technologies || [],
+      portfolio: user?.portfolio || "",
+      email: user?.email || "",
+    },
+  });
+  const {
+    fields: goalFields,
+    append: appendGoal,
+    remove: removeGoal,
+  } = useFieldArray<z.infer<typeof CompleteProfileEditSchema>>({
+    control,
+    name: "learningGoals",
+  });
+
+  const {
+    fields: experienceFields,
+    append: appendExperience,
+    remove: removeExperience,
+  } = useFieldArray<z.infer<typeof CompleteProfileEditSchema>>({
+    control,
+    name: "experiences",
+  });
+  const changeStep = async () => {
+    const nextStep = parseInt(currentStep) + 1;
+    const {
+      fullname,
+      portfolio,
+      availability,
+      learningGoals,
+      email,
+      experiences,
+      newProjects,
+    } = getValues();
+    try {
+      if (nextStep === 2) {
+        await updateUser({
+          updateData: {
+            fullname,
+            portfolio,
+            email,
+          },
+        });
+      }
+      if (nextStep === 3) {
+        await updateUser({
+          updateData: {
+            learningGoals,
+          },
+        });
+      }
+      if (nextStep === 4) {
+        await updateUser({
+          updateData: {
+            experiences: experiences?.map((experience) => experience.name),
+          },
+        });
+      }
+      if (nextStep === 5) {
+        startTransition(() => {
+          updateUser({
+            updateData: {
+              newProjects,
+              startTime: availability?.startTime,
+              endTime: availability?.endTime,
+            },
+          });
+        });
+      }
+    } catch (error) {
+      console.log(error);
+    }
+
+    setCurrentStep(nextStep.toString());
+  };
+  const [selectedFrom, setSelectedFrom] = React.useState<Date>();
+  const [selectedTo, setSelectedTo] = React.useState<Date>();
+
+  const handleDaySelectFrom = (date: Date | undefined) => {
+    if (!date) {
+      setSelectedFrom(date as any);
+      return;
+    }
+
+    const newDate = new Date(
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate()
+    );
+    setSelectedFrom(newDate as any);
+  };
+  const handleDaySelectTo = (date: Date | undefined) => {
+    if (!date) {
+      setSelectedTo(date as any);
+      return;
+    }
+
+    const newDate = new Date(
+      date.getFullYear(),
+      date.getMonth(),
+      date.getDate()
+    );
+    setSelectedTo(newDate as any);
+  };
```

Comment:
Can't this logic be reworked into the date components or the availability section itself? Right now you're prop drilling down multiple components when this component doesn't even use this function. 

https://www.geeksforgeeks.org/what-is-prop-drilling-and-how-to-avoid-it/